site/public/js/server.js

```diff
@@ -1388,6 +1398,27 @@ function deleteOverriddenGrades(user_id, g_id) {
     return false;
 }
 
+function confirmOverride(option, isDelete) {
+    $('.popup-form').css('display', 'none');
+    if (isDelete) {
+        deleteOverriddenGrades($('#user_id').val(), option);
+        $('#user_id').val('');
+    }
+    else {
+        $('input[name="option"]').val(option);
+        updateGradeOverride();
+        $('input[name="option"]').val(-1);
+    }
+}
+
+function overridePopup(json) {
+    $('.popup-form').css('display', 'none');
+    const form = $('#more_override_popup');
+    form[0].outerHTML = json['data']['popup'];
```

Comment:
It's very easy for jQuery functions which insert raw HTML to result in XSS vulnerabilities.  Please do not directly insert HTML this way.